docs/vignette_space.rst

```diff
@@ -0,0 +1,584 @@
+   ts = pyslim.SlimTreeSequence(
+         msprime.mutate(recap_ts, rate=1e-8, keep=True))
+
+   print(f"The tree sequence now has {ts.num_trees} trees,"
+         f" and {ts.num_mutations} mutations.")
+
+Note since :meth:`mutate <msprime.mutate>` is an msprime method, it does not return a pyslim
+tree sequence, so we need to convert it back.
+This has added mutations according to an infinite-sites model of mutation,
+resulting in
+
+.. code-block:: none
+
+   The tree sequence now has 16571 trees,and 31125 mutations.
+
+Two notes:
+We will have no further use for `slim_ts` or for `recap_ts`;
+we've just given them separate names for tidiness.
+And, since the original SLiM mutation had no mutations, we didn't need to specify `keep=True`
+in :meth:`mutate <msprime.mutate>`, but if we *had* put down selected mutations with SLiM
+we'd probably want to keep them around.
+
+
+****************************
+Take a sample of individuals
+****************************
+
+Ok, it's time to compute some things.
+In real life we don't get to work with *everyone* usually,
+so we'll take a subset of individuals.
+The range we have simulated has width and height 20 units,
+with a population density of around 1 per unit area.
+We'll get genomes to work with by pulling out
+
+1. All the modern individuals in the squares of width 4 in the corners of the range, and
+2. Five individuals sampled randomly from everyone alive 500 time steps ago.
+
+.. code-block:: python
+
+
+   np.random.seed(23)
+
+   alive = ts.individuals_alive_at(0)
+   locs = ts.individual_locations[alive, :]
+
+   cwidth = 3
+   corners = {
+      'topleft' : alive[np.logical_and(locs[:, 0] < cwidth, locs[:, 1] < 5)],
+      'topright' : alive[np.logical_and(locs[:, 0] < cwidth, locs[:, 1] > 15)],
+      'bottomleft' : alive[np.logical_and(locs[:, 0] > 20 - cwidth, locs[:, 1] < 5)],
+      'bottomright' : alive[np.logical_and(locs[:, 0] > 20 - cwidth, locs[:, 1] > 15)],
+      'center' : alive[np.logical_and(np.abs(locs[:, 0] - 10) < cwidth,
+                                      np.abs(locs[:, 1] - 10) < cwidth)]
+      }
+
+   old_ones = ts.individuals_alive_at(500)
+   corners['ancient'] = np.random.choice(old_ones, size=5)
+
+   for k in corners:
+      print(f"We have {len(corners[k])} individuals in the {k} group.")
+
+
+.. code-block:: none
+
+   We have 51 individuals in the topleft group.
+   We have 16 individuals in the topright group.
+   We have 22 individuals in the bottomleft group.
+   We have 35 individuals in the bottomright group.
+   We have 46 individuals in the center group.
+   We have 5 individuals in the ancient group.
+
+
+To keep names associated with each subset of individuals,
+we've kept the individuals in a dict, so that for instance
+`corners["topleft"]` is an array of all the individual IDs that are in the top left corner.
+The IDs of the ancient individuals we will work with are kept in the array `ancient`.
+
```

Comment:
Hey look you can make notes at specific locations. Like this one: here would be a good place to double-check that all the `ancient` samples are from a while ago, like:
```
for i in corners["ancient"]:
   ind = ts.individual(i)
   assert(ind.time >= 500 and ind.time < 520)
```